### PR TITLE
Recipe Application 

### PR DESCRIPTION
**Pull request by danielufeli**
Hello, 

Thanks for reviewing our project. 

Below are the tasks we implemented on this project:

- [x] #1
- [x] #2 
- [x] #3 
- [x] #4 
- [x] #5 
- [x] #6 
- [x] #8 
- [x] #9 
- [x] #10 
- [x] #21

### CODE REVIEW

File: app/controllers/recipe_foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+class RecipeFoodsController < ApplicationController
+  def create
+    food = Food.find_by(name: recipe_foods_params[:ingredient])
+    recipe = Recipe.find(recipe_foods_params[:recipe])
+    ingredient = RecipeFood.new(food:, recipe:, quantity: recipe_foods_params[:quantity])
```
Comment (FrederickMih) [inline app/controllers/recipe_foods_controller.rb:5]: - [x] I'm unable to add ingredients on the recipe show page. This console error occurs when I try to add food ingredients:

Screenshot from 2022-06-04 17-52-28

Kindly fix this issue.

File: app/views/recipes/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    <%= tag.span('Public:') %>
+    <label class="toggle__switch">
+      <%= check_box_tag 'public_recipe', 'yes', @recipe.public, disabled: true %>
+      <span class="slider round"></span>
+    </label>
+  </div>
+</div>
+<div class="recipe__buttons">
+  <% if can? :destroy, @recipe %>
+    <%= link_to 'Generate shopping list', '', class: 'btnn' %>
```
Comment (FrederickMih) [inline app/views/recipes/show.html.erb]: - [x] When I click on this link, no shopping list is generated. From the requirements of this project, we should be able to generate a shopping list that matches the URL `/general_shopping_list` when click on this link.

File: app/views/public_recipes/_public_recipe_card.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    <%
+    rf_f = recipe.recipe_foods.includes(:food)
+    rf_price = 0
+    rf_f.each { |item| rf_price += item.quantity * item.food.price }
+    %>
+    <%= tag.p("Total food items: #{rf_f.count}", class: 'card__text') %>
+    <%= tag.p("Total price: #{rf_price}", class: 'card__text') %>
+  </div>
+  <%= link_to '', recipe_path(recipe.id), id: "link_recipe-#{recipe.id}", class: 'card__hidden-link' %>
+</div>
```
Comment (FrederickMih) [inline app/views/public_recipes/_public_recipe_card.html.erb]: - [x] Please let's make sure that the page matches the structure of the given [wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/recipe-app/images/public_recipe_index.png?token=GHSAT0AAAAAABVGJOYEGHFXSNQCE4ZIK3DWYU3UQLQ). Total food items and total price show zero(0) as you can see on the screenshot below. Also, make sure that the total price is in **$**

Screenshot from 2022-06-04 18-24-15

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!




# To Highlights

- [ ] GitHub workflow has been followed :heavy_check_mark: 
- [ ] PR has a title and summary :heavy_check_mark: 

## Suggested changes

- [x] Please let's make sure our tests run without issues.

Screenshot from 2022-06-04 18-54-48

- [x] Please make sure to resolve the N + 1 queries problem from happening as requested in the project requirements

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by AshakaE — APPROVED** [review]
Hi @danielufeli , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**